runltp-ng

```diff
@@ -141,10 +144,12 @@ if ($run) {
 	my %results;
 	$results{'sysinfo'} = utils::collect_sysinfo($backend);
 	my ($stats, $test_results) =
-	    utils::run_ltp($backend, $ltpdir, $run, $timeout,
-			   $include, $exclude);
+		utils::run_ltp($backend, $ltpdir, $run, $timeout,
+			       $include, $exclude);
 	$results{'tests'} = {'stats' => $stats, 'results' => $test_results};
-	results::writelog(\%results, "$logname.json");
+	my $json_data = \%results;
+	$json_data = results::json_filter($json_filter, \%results) if $json_data;
+	results::writelog($json_data, "$logname.json");
```

Comment:
with above change, that `json_filter` is prepared for undefined `$json_filter`

```suggestion
	results::writelog(results::json_filter($json_filter, \%results), "$logname.json");
	results::writelog_html(\%results, "$logname.html");
```

Otherwise:
```suggestion
	my $json_data = \%results;
	$json_data = results::json_filter($json_filter, \%results) if $json_filter;
	results::writelog($json_data, "$logname.json");
```

Reply:
i am going for the first as this is also @metan-ucw idea in another comment